app/assets/javascripts/validation.coffee

```diff
@@ -99,17 +147,17 @@ $(document).ready ->
 
   getFieldType = (element) ->
     classes = $(element).attr('class').split(/\s+/)
-    if classes.indexOf('mmt-number') != -1
+    if classes.indexOf('mmt-number') != -1 or $(element).attr('number') == 'true'
       type = 'number'
-    if classes.indexOf('mmt-integer') != -1
+    if classes.indexOf('mmt-integer') != -1 or $(element).attr('integer') == 'true'
       type = 'integer'
-    if classes.indexOf('mmt-boolean') != -1
+    if classes.indexOf('mmt-boolean') != -1 or $(element).attr('boolean') == 'true'
       type = 'boolean'
-    if classes.indexOf('mmt-date-time') != -1
+    if classes.indexOf('mmt-date-time') != -1 or $(element).attr('date-time') == 'true'
       type = 'date-time'
-    if classes.indexOf('mmt-uri') != -1
+    if classes.indexOf('mmt-uri') != -1 or $(element).attr('uuid') == 'true'
```

Comment:
I think this is the wrong `attr`.
also, I know we've mentioned this but I don't think we've come to a definitive answer, do we want to use `prop` instead? https://stackoverflow.com/questions/5874652/prop-vs-attr